tests/unit_tests/test_suassystem_sda_converter.py

```diff
@@ -27,6 +27,63 @@ def setup_client_converter(self):
 
         self.sda_converter = SDAConverter(self.initial_coordinates, self.boundary_points)
 
+    def test_convert_fly_zones(self):
+
+        result_from_the_program=self.sda_converter.convert_fly_zones(numpy.array([self.boundary_points]))
+        test_location1=Location(self.boundary_points[0]["boundary_pts"][0]["latitude"],self.boundary_points[0]["boundary_pts"][0]["longitude"],0)
+        expected_result=(convert_to_point(self.initial_coordinates,test_location1)[:2])
+        self.assertEqual(expected_result[0],result_from_the_program[0][0][0])
+        self.assertEqual(expected_result[1],result_from_the_program[0][0][1])
+
+    def test_convert_fly_zone(self):
+
+        result_fly_zone=self.sda_converter.convert_fly_zone(self.boundary_points)
+        test_location1=Location(self.boundary_points[0]["boundary_pts"][0]["latitude"],self.boundary_points[0]["boundary_pts"][0]["longitude"],0)
+        expected_fly_zone=(convert_to_point(self.initial_coordinates,test_location1)[:2])
+        self.assertEqual(expected_fly_zone[0],result_fly_zone[0][0])
+        self.assertEqual(expected_fly_zone[1],result_fly_zone[0][1])
+
+    def test_is_obstacel_in_path(self):
+        test_waypoint = Location(38.8705508017, -77.3210866223, 91.44000244140625)
+        self.sda_converter.add_waypoint(test_waypoint)
```

Comment:
You should reset the waypoints before you add another waypoint. Additionally, I don't see the reason to use the ```add_waypoint``` method here instead of the ```set_current_waypoint``` method